base/reduce.jl

```diff
@@ -575,6 +575,13 @@ function any(f, itr)
     end
     return false
 end
+# @inbounds allows the compiler to optimize out the loop in case f is constant
+function any(f, a::AbstractArray)
```

Comment:
See #20469. In short, in the long term this will be safe, but for now it may have unintended effects for some custom types. Since this will only be a problem if they implement the iteration protocol incorrectly, I would say it's OK as a temporary situation, but feel free to argue otherwise on the other issue.